### PR TITLE
Coll/Hcoll: reduce_scatter(block) interface

### PR DESCRIPTION
**Pull request by vspetrov**

### CODE REVIEW

File: ompi/mca/coll/hcoll/coll_hcoll.h [hunk 1]
```diff
@@ -211,6 +213,18 @@ int mca_coll_hcoll_allreduce(const void *sbuf, void *rbuf, int count,
                             struct ompi_communicator_t *comm,
                             mca_coll_base_module_t *module);
```
Comment (jladd-mlnx) [inline ompi/mca/coll/hcoll/coll_hcoll.h:215]: Should be strictly greater than 4.5.

File: ompi/mca/coll/hcoll/coll_hcoll_module.c [hunk 2]
```diff
@@ -419,6 +424,12 @@ mca_coll_hcoll_comm_query(struct ompi_communicator_t *comm, int *priority)
     hcoll_module->super.coll_ialltoallv = hcoll_collectives.coll_ialltoallv ? mca_coll_hcoll_ialltoallv : NULL;
 #else
     hcoll_module->super.coll_ialltoallv = NULL;
+#endif
+#if HCOLL_API >= HCOLL_VERSION(4,5)
```
Comment (jladd-mlnx) [inline ompi/mca/coll/hcoll/coll_hcoll_module.c:428]: Strictly greater than v4.5

File: ompi/mca/coll/hcoll/coll_hcoll.h [hunk 3]
```diff
@@ -211,6 +213,18 @@ int mca_coll_hcoll_allreduce(const void *sbuf, void *rbuf, int count,
...
 
+#if HCOLL_API >= HCOLL_VERSION(4,5)
```
Comment (jladd-mlnx) [inline ompi/mca/coll/hcoll/coll_hcoll.h:216]: Should be strictly greater

File: ompi/mca/coll/hcoll/coll_hcoll_ops.c [hunk 4]
```diff
@@ -760,3 +763,88 @@ int mca_coll_hcoll_ialltoallv(const void *sbuf, int *scounts, int *sdisps,
     return rc;
 }
 #endif
+
```
Comment (jladd-mlnx) [inline ompi/mca/coll/hcoll/coll_hcoll_ops.c:766]: strictly greater than v4.5

File: ompi/mca/coll/hcoll/coll_hcoll_ops.c [hunk 5]
```diff
@@ -136,7 +136,9 @@ int mca_coll_hcoll_allgatherv(const void *sbuf, int scount,
                                              hcoll_module->previous_allgatherv_module);
```
Comment (jladd-mlnx) [inline ompi/mca/coll/hcoll/coll_hcoll_ops.c:136]: Is all this cosmetic stuff necessary?

File: ompi/mca/coll/hcoll/coll_hcoll.h [hunk 6]
```diff
@@ -302,15 +316,13 @@ int mca_coll_hcoll_ialltoall(const void *sbuf, int scount,
...
-int mca_coll_hcoll_ialltoallv(const void *sbuf, int *scounts,
-                            int *sdisps,
-                            struct ompi_datatype_t *sdtype,
-                            void *rbuf, int *rcounts,
-                            int *rdisps,
-                            struct ompi_datatype_t *rdtype,
-                            struct ompi_communicator_t *comm,
-                            ompi_request_t **req,
-                            mca_coll_base_module_t *module);
+int mca_coll_hcoll_ialltoallv(const void *sbuf, const int *scounts,
```
Comment (jladd-mlnx) [inline ompi/mca/coll/hcoll/coll_hcoll.h:319]: Cosmetic? Is it necessary?

Comment (vspetrov) [reply ompi/mca/coll/hcoll/coll_hcoll.h:319]: adds "const" - fixes compile warnings

**Comment by jladd-mlnx** [comment]
@vspetrov please cherry-pick into v4.1.x

**Comment by ibm-ompi** [comment]
The IBM CI (GNU/Scale) build failed! Please review the log, linked below.

 Gist: https://gist.github.com/ca53c7aeae846f1cb49f023b2345b6d4

**Comment by ibm-ompi** [comment]
The IBM CI (XL) build failed! Please review the log, linked below.

 Gist: https://gist.github.com/29777115aaaa604b71ee88f4663b8720

**Comment by ibm-ompi** [comment]
The IBM CI (PGI) build failed! Please review the log, linked below.

 Gist: https://gist.github.com/19408a76fc67516027f4f3168909c4fd

**Comment by vspetrov** [comment]
bot:retest

**Comment by vspetrov** [comment]
bot:retest

**Comment by awlauria** [comment]
bot:ompi:retest

**Comment by vspetrov** [comment]
bot:ompi:retest

**Comment by vspetrov** [comment]
> @vspetrov please cherry-pick into v4.1.x

@jladd-mlnx Done: https://github.com/open-mpi/ompi/pull/7902

**Comment by awlauria** [comment]
Any reason to not merge this? @jladd-mlnx

**Pull request merged**